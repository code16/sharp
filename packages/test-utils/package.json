{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}