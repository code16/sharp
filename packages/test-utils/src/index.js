export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}