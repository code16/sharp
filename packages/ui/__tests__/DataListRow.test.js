import merge from 'lodash/merge';

import SharpDataListRow from '../src/components/DataListRow.vue';
import { shallowMount } from '@vue/test-utils';


describe('DataListRow', ()=>{

    function createWrapper({ ...options }={}) {
        return shallowMount(SharpDataListRow, {
            ...options
        });
    }

    function withDefaultMocks(options) {
        return merge({
            computed: {
                hasLink: ()=>false
            },
            methods: {
                colClasses: ()=>'colClasses'
            },
        }, options);
    }

    test('mount', ()=>{
        const wrapper = createWrapper(withDefaultMocks());
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount header', ()=>{
        const wrapper = createWrapper(withDefaultMocks());
        wrapper.setProps({
            header: true
        });
        expect(wrapper.html()).toMatchSnapshot();
        wrapper.setProps({
            columns: [{ key:'label' }],
            row: { label: 'label' }
        });
        expect(wrapper.html()).toMatchSnapshot('columns');
    });

    test('mount with url', ()=>{
        const wrapper = createWrapper(withDefaultMocks({
            computed: {
                hasLink: ()=>true
            }
        }));
        wrapper.setProps({
            url: 'http://example.org'
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount append', ()=>{
        const wrapper = createWrapper(withDefaultMocks({
            slots: {
                append: 'mockedAppend'
            }
        }));
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount columns', ()=>{
        const wrapper = createWrapper(withDefaultMocks());
        wrapper.setProps({
            columns: [{ key:'name' }, { key:'age', html: true }],
            row: { name:'John', age:'<span>3</span>' }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('mount cell slot', ()=>{
        const wrapper = createWrapper(withDefaultMocks({
            scopedSlots: {
                cell: '<div v-bind="props">{{ props.column.key }}:{{ props.row.name }}</div>'
            }
        }));
        wrapper.setProps({
            columns: [{ key:'name' }],
            row: { name:'John' },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    describe('computed', ()=>{
        test('hasUrl', ()=>{
            const wrapper = createWrapper();
            expect(wrapper.vm.hasLink).toBe(false);
            wrapper.setProps({
                url: 'http://example.org',
            });
            expect(wrapper.vm.hasLink).toBe(true);
        });
    });

    describe('methods', ()=>{
        test('colClasses', ()=>{
            const wrapper = createWrapper();
            expect(wrapper.vm.colClasses({ size:6, sizeXS:12 }))
                .toEqual(expect.arrayContaining([
                    'col-12',
                    'col-md-6',
                ]));
            expect(wrapper.vm.colClasses({ hideOnXS:true }))
                .toContain('d-none d-md-flex');
        });
    });
});