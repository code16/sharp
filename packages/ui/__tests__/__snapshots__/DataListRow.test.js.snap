// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataListRow mount 1`] = `
<div class="SharpDataList__row container SharpDataList__row--disabled">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3"></div>
		<!---->
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;

exports[`DataListRow mount append 1`] = `
<div class="SharpDataList__row container SharpDataList__row--disabled">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3"></div>
		<!---->
	</div>
	<div class="SharpDataList__row-append align-self-center">mockedAppend</div>
</div>
`;

exports[`DataListRow mount cell slot 1`] = `
<div class="SharpDataList__row container SharpDataList__row--disabled">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3">
			<div class="px-2 px-md-3 SharpDataList__td colClasses">
				<div row="[object Object]" column="[object Object]">name:John</div>
			</div>
		</div>
		<!---->
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;

exports[`DataListRow mount columns 1`] = `
<div class="SharpDataList__row container SharpDataList__row--disabled">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3">
			<div class="px-2 px-md-3 SharpDataList__td colClasses">
				John
			</div>
			<div class="px-2 px-md-3 SharpDataList__td colClasses">
				<div class="SharpDataList__td-html-container">
					<span>3</span>
				</div>
			</div>
		</div>
		<!---->
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;

exports[`DataListRow mount header 1`] = `
<div class="SharpDataList__row container SharpDataList__row--header">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3"></div>
		<!---->
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;

exports[`DataListRow mount header: columns 1`] = `
<div class="SharpDataList__row container SharpDataList__row--header">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3">
			<div class="px-2 px-md-3 SharpDataList__th colClasses">
				label
			</div>
		</div>
		<!---->
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;

exports[`DataListRow mount with url 1`] = `
<div class="SharpDataList__row container">
	<div class="SharpDataList__cols">
		<div class="row mx-n2 mx-md-n3"></div>
		<a class="SharpDataList__row-link" href="http://example.org"></a>
	</div>
	<div class="SharpDataList__row-spacer"></div>
</div>
`;
